frontend2/src/components/CanvasActions/index.tsx

```diff
@@ -83,12 +83,20 @@ export function CanvasActions() {
       });
       toast({ id: "add-success", variant: "success", content: "Added!" });
       if (!useCanvasState.getState().isAdmin) setCoolDownLeft(5);
-    } catch {
-      toast({
-        id: "add-failure",
-        variant: "error",
-        content: "Error occurred. Please check your wallet connection and try again.",
-      });
+    } catch (e) {
+      if (typeof e === 'string') {
+        toast({
+          id: "add-failure",
+          variant: "error",
+          content: `Wallet Error: ${e}.`,
```

Comment:
Did you test locally that `e` will actually be a string here and not require something like `e.message`? Also, how long can these error messages get? I would like to avoid a scenario where we create a giant toast message that fills the entire screen.

Reply:
The errors returned from IC currently are strings (we are going to follow up on making them better). All the errors I've seen are very short.